### PR TITLE
Adapts dashboard to different challenge logic + cleans JS code

### PR DESCRIPTION
**Pull request by vdequercize**
In this update : 
- The dashboards now takes into account the different challenges logic
- The % can't go over 100%
- I cleaned the JS code

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -70,8 +69,6 @@ def handle_activities
       activity_to_delete = Activity.find_by strava_id: params["object_id"]
       activity_to_delete.destroy
       puts "Activity #{params['object_id']} has been destroyed"
-    else
     end
   end
-
```
Comment (vdequercize) [inline app/controllers/pages_controller.rb:76]: Cleaned up unecessary white spaces

File: app/javascript/controllers/activity_controller.js [hunk 2]
```diff
@@ -11,15 +11,11 @@ export default class extends Controller {
...
-     console.log(type.value)
    }
   cycling() {
      console.log("cycling")
      this.activityTarget.value="CYCLING"
      this.cyclingTarget.classList.add("toggle")
      this.runningTarget.classList.remove("toggle")
-     let type = document.querySelector("#challenge_activity_type")
-     console.log(type.value)
      }
```
Comment (vdequercize) [inline app/javascript/controllers/activity_controller.js:20]: Cleaned up useless JS lines (used for development)

File: app/javascript/controllers/flatpickr_controller.js [hunk 3]
```diff
@@ -5,12 +5,10 @@ import flatpickr from "flatpickr";
...
   // Inform the controller that it has two targets in the form, which are our inputs
-  static targets = [ "startTime", "endTime", "startTime2", "endTime2" ]
+  static targets = [ "startTime", "endTime" ]
 
   connect() {
     flatpickr(this.startTimeTarget, {})
     flatpickr(this.endTimeTarget, {})
-    flatpickr(this.startTime2Target, {})
-    flatpickr(this.endTime2Target, {})
   }
```
Comment (vdequercize) [inline app/javascript/controllers/flatpickr_controller.js:13]: Cleaned up useless JS lines (used for development)

File: app/views/pages/dashboard.html.erb [hunk 4]
```diff
@@ -72,24 +72,39 @@
...
                 <p style="text-align: left;">Distance achieved</p>
                 <% distance_progress = number_to_percentage((c.all_activities_km / c.target_distance)*100, precision: 2) %>
-                <div class="progress ">
+                <% else %>
+                <p style="text-align: left;">Sessions achieved</p>
+                <% distance_progress = number_to_percentage((c.all_activities_sessions / c.target_sessions)*100, precision: 2) %>
+                <% end %>
                   <div class="progress-bar bg-green" role="progressbar" style="width: <%= distance_progress %>;" aria-valuenow="25" aria-valuemin="0" aria-valuemax="100"><%= distance_progress %></div>
                 </div>
               </div>
```
Comment (vdequercize) [inline app/views/pages/dashboard.html.erb:110]: Updates dashboard to take into account different challenges logic and rephrased the text

**Comment by jean852** [comment]
Noice !

**Pull request merged**